staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -284,27 +282,32 @@ func (s *SpdyRoundTripper) tlsConn(ctx context.Context, rwc net.Conn, targetHost
 // dialWithoutProxy dials the host specified by url, using TLS if appropriate.
 func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(url)
+	netD := s.Dialer
+	if netD == nil {
+		netD = &net.Dialer{
+			Timeout: 30 * time.Second,
```

Comment:
This matches timeout used on line 222: https://github.com/kubernetes/kubernetes/pull/109651/files#diff-be99fb028cbdec52a7ee54d6b9a97c60a76f288124640e5f41a6aa96807fb8bbL222